<?xml version="1.0" encoding="utf-8"?>
<launch>

  <group ns="/wallframe/core">
    <!-- set param -->
    <arg name="app_manifest_path"
         default="$(find wallframe_core)/wallframe_application_manifest.yaml" />
    <param name="paths/app_manifest" textfile="$(arg app_manifest_path)" />
    <param name="paths/application_path" value="/home/yifan/wallframe_workspace/wallframe_apps/" /> 

    <!-- create app_manager node -->
    <node pkg="wallframe_core" name="wallframe_app_manager"
          type="wallframe_app_manager_new.py" output="screen"/>
    
    <node pkg="wallframe_extra" name="wallframe_simple_mouse"
          type="wallframe_simple_mouse.py" output="screen" />
  </group>
  
  <!-- Infobar at the bottom of the screen that shows user avatar and messages -->
  <!-- <node pkg="wallframe_core" name="wallframe_infobar" type="wallframe_infobar.py" output="screen"/> -->
  <!-- Parameters for the application menu -->
  <param name="/wallframe/menu/params/cursor_path" value="$(find wallframe_core)/assets/cursor_open.png" /> 
  <param name="/wallframe/menu/params/cursor_path_alt" value="$(find wallframe_core)/assets/cursor_closed.png" /> 
  <param name="/wallframe/menu/params/background_path" value="$(find wallframe_core)/assets/background.jpg" />
  <!-- Run the application menu -->
  <node pkg="wallframe_core" name="wallframe_app_tileflow_menu" type="wallframe_app_tileflow_menu.py" output="screen"/>

  <!-- yaml format -->
  <!-- x: x offset y: y offset -->
  <!-- width/height: screen width/height -->
  <rosparam>
    wallframe:
      core:
        params:
          x: 0
          y: 0
          width: 1400
          height: 900
          cursor_width: 50
          cursor_height: 50
        default_app: image_storm
      infobar:
        params:
          height_percentage: 0.03        
          icon_width: 5
          icon_spacer: 5
          text_height: 2
      menu:
        params:
          height_percentage: .97
          workspace_size: [-1200,1200,-600,600,1500,3000]
          border_scale: .02
          y_offset: -600.0
          screensaver: no
  </rosparam>

</launch>
